File: src/main/java/gargoyle/ct/prop/impl/simple/CTSimpleEnumProperty.java
package gargoyle.ct.prop.impl.simple;

import org.jetbrains.annotations.NotNull;

public class CTSimpleEnumProperty<E extends Enum<E>> extends CTSimpleProperty<E> {
    public CTSimpleEnumProperty(Class<E> type, String name) {
        super(type, name);
    }

    public CTSimpleEnumProperty(Class<E> type, String name, @NotNull E def) {
        super(name, def);
    }

    public CTSimpleEnumProperty(@NotNull Class<E> type) {
        super(type, type.getSimpleName());
    }

    public CTSimpleEnumProperty(Class<E> type, @NotNull E def) {
        super(type.getSimpleName(), def);
    }

    public CTSimpleEnumProperty(String name, @NotNull E def) {
        super(name, def);
    }

    public CTSimpleEnumProperty(@NotNull E def) {
        super(def.getClass().getSimpleName(), def);
    }
}

File: src/main/kotlin/gargoyle/ct/prop/impl/simple/CTSimpleEnumProperty.kt
package gargoyle.ct.prop.impl.simple

import kotlin.reflect.KClass

class CTSimpleEnumProperty<E : Enum<E>> : CTSimpleProperty<E> {
    constructor(type: KClass<E>, name: String) : super(type, name)
    constructor(type: KClass<E>) : super(type, type.qualifiedName!!.substringAfterLast('.'))
    constructor(type: KClass<E>, def: E) : super(type.qualifiedName!!.substringAfterLast('.'), def)
    constructor(name: String, def: E) : super(name, def)
    constructor(def: E) : super(def.javaClass.simpleName, def)
}
